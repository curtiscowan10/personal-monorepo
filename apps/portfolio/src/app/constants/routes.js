/**
 * Route path constants for the portfolio application
 */
export const ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  PROJECTS: '/projects',
  CONTACT: '/contact',
};

/**
 * Route configuration with metadata
 */
export const ROUTE_CONFIG = {
  [ROUTES.HOME]: {
    path: ROUTES.HOME,
    name: 'Home',
    title: 'Home',
  },
  [ROUTES.ABOUT]: {
    path: ROUTES.ABOUT,
    name: 'About',
    title: 'About Me',
  },
  [ROUTES.PROJECTS]: {
    path: ROUTES.PROJECTS,
    name: 'Projects',
    title: 'My Projects',
  },
  [ROUTES.CONTACT]: {
    path: ROUTES.CONTACT,
    name: 'Contact',
    title: 'Contact Me',
  },
};
