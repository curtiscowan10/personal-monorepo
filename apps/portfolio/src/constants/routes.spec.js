import { ROUTES, ROUTE_CONFIG } from './routes';

describe('routes constants', () => {
  describe('ROUTES', () => {
    it('should export correct route paths', () => {
      expect(ROUTES.HOME).toBe('/');
      expect(ROUTES.ABOUT).toBe('/about');
      expect(ROUTES.PROJECTS).toBe('/projects');
      expect(ROUTES.CONTACT).toBe('/contact');
    });

    it('should have all required routes defined', () => {
      expect(ROUTES).toHaveProperty('HOME');
      expect(ROUTES).toHaveProperty('ABOUT');
      expect(ROUTES).toHaveProperty('PROJECTS');
      expect(ROUTES).toHaveProperty('CONTACT');
    });

    it('should have string values for all routes', () => {
      Object.values(ROUTES).forEach((route) => {
        expect(typeof route).toBe('string');
        expect(route.length).toBeGreaterThan(0);
      });
    });

    it('should have unique route paths', () => {
      const routeValues = Object.values(ROUTES);
      const uniqueValues = [...new Set(routeValues)];
      expect(routeValues).toHaveLength(uniqueValues.length);
    });
  });

  describe('ROUTE_CONFIG', () => {
    it('should have configuration for all routes', () => {
      Object.keys(ROUTES).forEach((routeKey) => {
        const routePath = ROUTES[routeKey];
        expect(ROUTE_CONFIG).toHaveProperty(routePath);
      });
    });

    it('should have correct home route configuration', () => {
      const homeConfig = ROUTE_CONFIG[ROUTES.HOME];
      expect(homeConfig).toEqual({
        path: '/',
        name: 'Home',
        title: 'Home',
      });
    });

    it('should have correct about route configuration', () => {
      const aboutConfig = ROUTE_CONFIG[ROUTES.ABOUT];
      expect(aboutConfig).toEqual({
        path: '/about',
        name: 'About',
        title: 'About Me',
      });
    });

    it('should have correct projects route configuration', () => {
      const projectsConfig = ROUTE_CONFIG[ROUTES.PROJECTS];
      expect(projectsConfig).toEqual({
        path: '/projects',
        name: 'Projects',
        title: 'My Projects',
      });
    });

    it('should have correct contact route configuration', () => {
      const contactConfig = ROUTE_CONFIG[ROUTES.CONTACT];
      expect(contactConfig).toEqual({
        path: '/contact',
        name: 'Contact',
        title: 'Contact Me',
      });
    });

    it('should have consistent paths between ROUTES and ROUTE_CONFIG', () => {
      Object.entries(ROUTES).forEach(([key, path]) => {
        expect(ROUTE_CONFIG[path].path).toBe(path);
      });
    });

    it('should have all required properties for each route config', () => {
      Object.values(ROUTE_CONFIG).forEach((config) => {
        expect(config).toHaveProperty('path');
        expect(config).toHaveProperty('name');
        expect(config).toHaveProperty('title');
        expect(typeof config.path).toBe('string');
        expect(typeof config.name).toBe('string');
        expect(typeof config.title).toBe('string');
      });
    });

    it('should match snapshot', () => {
      expect({ ROUTES, ROUTE_CONFIG }).toMatchSnapshot();
    });
  });
});
