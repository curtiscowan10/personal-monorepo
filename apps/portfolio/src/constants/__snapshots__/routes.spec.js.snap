// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`routes constants > ROUTE_CONFIG > should match snapshot 1`] = `
{
  "ROUTES": {
    "ABOUT": "/about",
    "CONTACT": "/contact",
    "HOME": "/",
    "PROJECTS": "/projects",
  },
  "ROUTE_CONFIG": {
    "/": {
      "name": "Home",
      "path": "/",
      "title": "Home",
    },
    "/about": {
      "name": "About",
      "path": "/about",
      "title": "About Me",
    },
    "/contact": {
      "name": "Contact",
      "path": "/contact",
      "title": "Contact Me",
    },
    "/projects": {
      "name": "Projects",
      "path": "/projects",
      "title": "My Projects",
    },
  },
}
`;
